microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -79,6 +89,60 @@ def run_tests(
 
         assert_that(failures, f"Unexpected failures: {failures}").is_empty()
 
+    def run_metrics_tests(
+        self,
+        test_result: TestResult,
+        environment: Environment,
+        hypervisor: str,
+        log_path: Path,
+        skip: Optional[List[str]] = None,
+    ) -> None:
+        self.per_mtr_report_file = log_path.joinpath("perf_metrics.json")
+
+        perf_metrics_tests = self._list_perf_metrics_tests(hypervisor=hypervisor)
+
+        for testcase in perf_metrics_tests:
+            testcase_result = CHPerfMetricTestResult()
+            testcase_result.name = testcase
+
+            try:
+                result = self.run(
+                    f"tests --hypervisor {hypervisor} --metrics -- -- \
+                        --test-filter {testcase}",
+                    timeout=self.TIME_OUT,
+                    force_run=True,
+                    cwd=self.repo_root,
+                    no_info_log=False,  # print out result of each test
+                    shell=True,
+                )
+                output = result.stdout.replace("\r\n", "\n")
+                output = output.replace("\t", "")
+                if result.exit_code == 0:
+                    metrics = self._process_perf_metric_test_result(result.stdout)
+                    testcase_result.status = TestStatus.PASSED
+                    testcase_result.metrics = metrics
```

Comment:
The intermediate variable `metrics` is not necessary.

Reply:
Removed

Reply:
It's not removed, As my previous comments, it can be like
```python
                if result.exit_code == 0:
                    testcase_result.status = TestStatus.PASSED
                    testcase_result.metrics = self._process_perf_metric_test_result(result.stdout)

```

Reply:
Done